sources/platform/actors/development/actor_definition/dataset_schema/validation.md

```diff
@@ -0,0 +1,223 @@
+            }]
+        }
+    }
+}
+```
+
+The type of the AJV validation error object is [here](https://github.com/ajv-validator/ajv/blob/master/lib/types/index.ts#L86)
+
+If you use the Apify JS client or Apify SDK and call `pushData` function you can access the validation errors in a `try catch` block like this:
+
+```javascript
+try {
+    const response = await Actor.pushData(items);
+} catch (error) {
+    if (!error.data?.invalidItems) throw error;
+    error.data.invalidItems.forEach((item) => {
+        const { itemPosition, validationErrors } = item;
+    });
+}
+```
+
+## Examples
+
+Optional field (price is optional in this case):
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string"
+        },
+        "price": {
+                "type": "number"
+        }
+    },
+    "required": ["name"]
+}
+```
+
+Field with multiple types:
+
+```json
+{
+    "price": {
+        "type": ["string", "number"]
+    }
+}
+```
+
+Field with type `any`:
+
+```json
+{
+    "price": {
+        "type": ["string", "number", "object", "array", "boolean"]
+    }
+}
+```
+
+Enabling fields to be `null` :
+
+```json
+{
+    "name": {
+        "type": "string",
+        "nullable": true
+    }
+}
+```
+
+Define type of objects in array:
+
+```json
+{
+    "comments": {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "author_name": {
+                    "type": "string"
+                }
+            }
+        }
+    }
+}
+```
+
+Define specific fields, but allow anything else to be added to the item:
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string"
+        }
+    },
+    "additionalProperties": true
+}
+```
+
+See [json schema reference](https://json-schema.org/understanding-json-schema/reference) for additional options.
+
+Example of schema generator [here](https://www.liquid-technologies.com/online-json-to-schema-converter).
+
+# Dataset field statistics
+
+When you have the dataset fields schema set up, we then use the schema to generate a list of fields and measure statistics for these fields.
+
+The measured statistics are following:
+
+- **Null count:** how many items in the dataset have the field set to null
+- **Empty count:** how many items in the dataset are `undefined` , meaning that for example empty string is not considered empty
+- **Minimum and maximum**
+  - For numbers, this is calculated directly
+  - For strings, this field tracks string length
+  - For arrays, this field tracks the number of items in the array
+  - For objects, this tracks the number of keys
+
+:::note
+
+Currently, you cannot view these statistics. We will add API endpoint soon. But you can already use them in monitoring.
```

Comment:
The `monitoring` label should be a link to the section of the monitoring docs talking about the statistics.